packages/core/src/components/forms/_file-input.scss

```diff
@@ -64,6 +64,16 @@ $file-input-button-padding-small: ($pt-input-height-small - $pt-button-height-sm
         }
       }
     }
+
+    &:focus + .#{$ns}-file-upload-input {
+      box-shadow: input-transition-shadow($input-shadow-color-focus, true),
+        $input-box-shadow-focus;
+
+      .#{$ns}-dark & {
+        box-shadow: dark-input-transition-shadow($dark-input-shadow-color-focus, true),
+          $pt-dark-input-box-shadow;
+      }
+    }
```

Comment:
is this a paired down version of the `pt-input-intent` mixin? maybe we could comment that if so to at least loosely link these

also wondering if we should be using the same `2px` found in that mixin for consistency? is there a reason these should be different?

Reply:
The box shadows here were pulled from the base `pt-input` mixin:
https://github.com/palantir/blueprint/blob/dc1b16237900c9b49e37f31b7a95523c16ae07e0/packages/core/src/components/forms/_common.scss#L109-L112 https://github.com/palantir/blueprint/blob/dc1b16237900c9b49e37f31b7a95523c16ae07e0/packages/core/src/components/forms/_common.scss#L59-L73 https://github.com/palantir/blueprint/blob/c50a4a49b736b87dfa36284565b6975ef0b00dbb/packages/core/src/common/_mixins.scss#L100-L102 https://github.com/palantir/blueprint/blob/dc1b16237900c9b49e37f31b7a95523c16ae07e0/packages/core/src/components/forms/_common.scss#L30-L32
and the base `pt-dark-input` mixin:
https://github.com/palantir/blueprint/blob/dc1b16237900c9b49e37f31b7a95523c16ae07e0/packages/core/src/components/forms/_common.scss#L200-L202 https://github.com/palantir/blueprint/blob/dc1b16237900c9b49e37f31b7a95523c16ae07e0/packages/core/src/components/forms/_common.scss#L75-L89

The default focus box shadow on input groups without an intent appears to be 2px total (1px inset + 1px).
![Screenshot 2024-08-28 at 11 10 20@2x](https://github.com/user-attachments/assets/c26415fd-a6bc-4abc-8260-93f81a2e9fd4)

For input groups with an intent, it appears to be 3px (1px inset + 2px)
![Screenshot 2024-08-28 at 11 10 11@2x](https://github.com/user-attachments/assets/86fff015-04ca-4e71-adc7-9371d6d640f1)

I chose to go with the non-intent style box shadow for file input since it currently does not have the ability to have our intent styles applied. I do think it's a bit odd that the default, non-intent focus outline doesn't match the width of the focused outlines. Perhaps that's something to address in a FLUP? For now, I can comment to reference where these are coming from.

Reply:
> I do think it's a bit odd that the default, non-intent focus outline doesn't match the width of the focused outlines.

ahh nice catch - I just recently updated these focus styles. looking back at this comment: https://github.com/palantir/blueprint/pull/6854#issuecomment-2200929002 I'm seeing originally I used a lighter color for the 2px which then became the same color (and a that point I probably just missed keeping it consistent at 2px)

Reply:
oh - maybe I did that because it already starts out with the 1px colored border when not focused

agreed without intent it makes sense for this to just be the 2px focus state!